data_structures/binary_tree/binary_search_tree.py

```diff
@@ -34,7 +34,14 @@ def __str__(self) -> str:
     def __reassign_nodes(self, node: Node, new_children: Node | None) -> None:
         if new_children is not None:  # reset its kids
             new_children.parent = node.parent
-        if node.parent is not None:  # reset its parent
+        if (
+            node == self.root
+        ):  # This check if the node is the head and assign the root to the remaining
```

Comment:
Please do not wrap line for a comment.
```suggestion
        # This check if the node is the head and assigns the root to the remaining
        if node == self.root:
```